In [9]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [10]:
immoSV_train = pd.read_csv(r'.\data\8it\8_immoSV_train.csv',delimiter=',', decimal='.')
immoSV_test = pd.read_csv(r'.\data\8it\8_immoSV_test.csv',delimiter=',', decimal='.')

In [11]:
y_train = immoSV_train["median_house_value"]
X_train = immoSV_train.drop(["median_house_value"], axis=1)
y_test = immoSV_test["median_house_value"]
X_test = immoSV_test.drop(["median_house_value"], axis=1)

In [12]:
# Fit regression model (using the natural log of one of the regressors)
results = smf.ols('y_train ~ X_train', data=immoSV_train).fit()

In [13]:
# Inspect the results
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                y_train   R-squared:                       0.649
Model:                            OLS   Adj. R-squared:                  0.649
Method:                 Least Squares   F-statistic:                     1922.
Date:                Fri, 24 Sep 2021   Prob (F-statistic):               0.00
Time:                        12:37:03   Log-Likelihood:            -1.4345e+05
No. Observations:               11428   AIC:                         2.869e+05
Df Residuals:                   11416   BIC:                         2.870e+05
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept    2.611e+05   1.08e+04     24.282      

R² :  0.649. Les variables de notre modèle explique 65% environ de la variance du prix médian.

coef : les valeurs des paramètres du modèle, les coefficients pour chacune des 1à variables et l'intercept.

std err : erreur standard.

P>|t| : p-value. Ici elle sont toutes inférieures à 0.05, sauf X_train[8] qui est NEAR_OCEAN. Pour ces variables, on rejete l'hypothèse nulle, elles ont donc un effet sur la target. Pour NEAR_OCEAN .

[0.025  0.975] : intervalle de confiance à 95%. 